build/index.js

```diff
@@ -0,0 +1,127 @@
+      })
+      .then(function () {
+        return xcodeUtil.getXcodeProject(xcodeProjectPath);
+      })
+      .then(function (xcodeProject) {
+        var project = xcodeProject._project,
+          firstProject = project.getFirstProject().firstProject,
+          firstTarget = project.getFirstTarget().firstTargetUuid;
+
+        target = project.addTarget('widget', 'app_extension', 'widget');
+        // create widget group
+        groupKey = project.pbxCreateGroup('widget', 'widget');
+        // Add newly created widget group to the main group
+        project.addToPbxGroup({
+          fileRef: groupKey,
+          basename: 'widget'
+        }, firstProject.mainGroup);
+
+        // add widget as a build dependency for the app
+        project.addTargetDependency(firstTarget, [target]);
+
+        return xcodeProject;
+      })
+      .then(function (xcodeProject) {
+        var project = xcodeProject._project;
+
+        // Add files
+        module_config.code.forEach(function(file) {
+          if (isHeaderFile(file)) {
+            project.addHeaderFile(file, {target: target.uuid}, groupKey);
+          } else if (isSourceFile(file)) {
+            project.addSourceFile(file, {target: target.uuid, ext: true}, groupKey);
+          } else {
+            console.warn('Skipping unknown code file type', file);
+          }
+        });
+
+        // Add resources
+        module_config.resources.forEach(function(resource) {
+          xcodeProject._project.addResourceFile(resource, {target: target.uuid, ext: true}, 'widget');
+        });
+
+        // Add frameworks
+        module_config.frameworks.forEach(function(framework) {
+          xcodeProject._project.addFramework(framework, {link: true, target: target.uuid, ext: true});
+        });
+        return xcodeProject;
+      })
+      .then(function (xcodeProject) {
+        xcodeProject.write();
+      })
+      .then(function () {
+        // Update plist and entitlements
+        // Teleaf entitlements
+        var entitlements = updatePlist.get(path.join(xcodeProjectPath, 'TeaLeafIOS.entitlements'));
+        var rawEntitlements = entitlements.getRaw();
+        rawEntitlements['com.apple.security.application-groups'] = [widgetGroup];
+
+        // Widget Plist
+        var widgetPlist = updatePlist.getInfoPlist(path.join(xcodeProjectPath, 'widget'));
+        var raw = widgetPlist.getRaw();
+        raw.CFBundleDisplayName = app.manifest.title || "";
+        raw.CFBundleIdentifier = config.bundleID + '.' + widgetID;
+        raw.CFBundleShortVersionString = version;
+        raw.CFBundleVersion = buildNumber + "";
+        raw.WidgetGroup = widgetGroup;
+
+        // Widget Entitlements
+        var widgetEntitlements = updatePlist.get(path.join(config.xcodeProjectPath, 'widget', 'widget.entitlements'));
+        var rawWidgetEntitlements = widgetEntitlements.getRaw();
+        rawWidgetEntitlements['com.apple.security.application-groups'] = [widgetGroup];
```

Comment:
same comment as above,

This will avoid having these values in the plist file too